source/base/polynomial.cc

```diff
@@ -1228,6 +1234,230 @@ namespace Polynomials
+          }
+        else if (index==1)
+          {
+            this->lagrange_support_points[0] = 0.;
+            this->lagrange_support_points[1] = 1.;
+            this->lagrange_support_points[2] = 1.;
+            this->lagrange_weight = 6.75;
+          }
+        else if (index==2)
+          {
+            this->lagrange_support_points[0] = 0.;
+            this->lagrange_support_points[1] = 0.;
+            this->lagrange_support_points[2] = 1.;
+            this->lagrange_weight = -6.75;
+          }
+        else if (index==3)
+          {
+            this->lagrange_support_points[0] = 0.;
+            this->lagrange_support_points[1] = 0.;
+            this->lagrange_support_points[2] = 5./7.;
+            this->lagrange_weight = 3.5;
+          }
+      }
+
+    // Higher order Polynomials degree>=4: the entries (1,0) and
+    // (degree,degree-1) of the mass matrix will be equal to 0
+    //
+    //     | x  0  x  x         x  x  x |
+    //     | 0  x  x  x  . . .  x  x  x |
+    //     | x  x  x  x         x  x  x |
+    //     | x  x  x  x         x  x  x |
+    //     |     .       .         .    |
+    // M = |     .         .       .    |
+    //     |     .           .     .    |
+    //     | x  x  x  x         x  x  x |
+    //     | x  x  x  x  . . .  x  x  0 |
+    //     | x  x  x  x         x  0  x |
+    //
+    if (degree >= 4)
+      {
+        // We find the inner points as the zeros of the Jacobi polynomials
+        // with alpha = beta = 2 which is the polynomial with the kernel
+        // (1-x)^2 (1+x)^2, the two polynomials achieving zero value and zero
+        // derivative at the boundary. The zeros of the Jacobi polynomials are
+        // given by the eigenvalues to a symmetric tridiagonal matrix with the
+        // entries given below. For degree 4 the eigenvalue is zero, so bypass
+        // the LAPACK logic in that case. Note that we compute the points for
+        // the interval [-1,1], so we must scale the final eigenvalues in the
+        // code below.
+
+        Vector<double> eigenvalues;
+        if (degree > 4)
+          {
+            LAPACKFullMatrix<double> jacobi_support_points_mat(degree-3,
+                                                               degree-3);
+            for (unsigned int k=1; k<degree-3; k++)
+              {
+                jacobi_support_points_mat(k-1,k) =
+                  std::sqrt(4.*k*(k+2.)*(k+2.)*(k+4.)/((2.*k+3.)*(2.*k+4.)*(2.*k+4.)*(2.*k+5.)));
+                jacobi_support_points_mat(k,k-1) = jacobi_support_points_mat(k-1,k);
+              }
+
+            // calculate the eigenvalues = zero points of the Jacobi polynomials
+            FullMatrix<double> eigenvectors(degree-3,degree-3);
+            jacobi_support_points_mat.compute_eigenvalues_symmetric(-1., 1., 1.e-20,
+                                                                    eigenvalues,
+                                                                    eigenvectors);
```

Comment:
I would prefer to rescale the roots here instead of doing that at all the places where they are used.

Reply:
That's a good idea. I've fixed it now.